### PR TITLE
[3] Add PL config, commands , entity, persistence and tests

### PR DESCRIPTION
**Pull request by Yurchenko-Anton**

### CODE REVIEW

File: src/main/java/com/example/matcher/config/JsonAuditRecordPublisher.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    public void publish(Stream<? extends AuditRecord> auditRecords) {
+        final String auditRecordsStr;
+        try {
+            auditRecordsStr = objectMapper.writeValueAsString(auditRecords);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        System.out.println(auditRecordsStr);
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/config/JsonAuditRecordPublisher.java:24]: remove it please

File: src/main/java/com/example/matcher/repository/LocationPersistence.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.Collection;
+
+@Component
+public class LocationPersistence {
+
+    private final PersistenceLayer persistenceLayer;
+    private final PLContext settings;
+
+    public LocationPersistence(PLContext settings) {
```
Comment (asinchuk) [inline src/main/java/com/example/matcher/repository/LocationPersistence.java:16]: Lombok?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
 server.port=8083
-spring.datasource.url=jdbc:mysql://localhost:3300/security?createDatabaseIfNotExist=true
+spring.datasource.url=jdbc:mysql://localhost:3300/security?createDatabaseIfNotExist=true&rewriteBatchedStatements=true
```
Comment (asinchuk) [inline src/main/resources/application.properties:2]: is this argument necessary?

File: src/test/java/com/example/matcher/repository/LocationPersistenceTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    static void deleteTestData() {
+        plContext.dslContext()
+                .delete(Locations.TABLE)
+                .where(DSL.field(LocationsEntity.STREET_NAME.toString()).eq(UPDATE_STREET_NAME))
+                .execute();
+    }
+
+    @Test
+    void shouldUpdateData() {
+        var queryBeforeUpdate = plContext
```
Comment (asinchuk) [inline src/test/java/com/example/matcher/repository/LocationPersistenceTest.java:54]: final var

File: src/test/java/com/example/matcher/repository/LocationPersistenceTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                .select(LocationsEntity.ID)
+                .from(LocationsEntity.INSTANCE)
+                .where(LocationsEntity.STREET_NAME.eq(START_STREET_NAME))
+                .fetch();
+
+        AtomicInteger id = new AtomicInteger();
```
Comment (asinchuk) [inline src/test/java/com/example/matcher/repository/LocationPersistenceTest.java:60]: avoid it please

File: src/main/resources/application.properties (hunk 3 above)
Comment (Yurchenko-Anton) [reply src/main/resources/application.properties:2]: yes

File: src/main/java/com/example/matcher/repository/LocationPersistence.java (hunk 2 above)
Comment (Yurchenko-Anton) [reply src/main/java/com/example/matcher/repository/LocationPersistence.java:16]: Unreal

**Review by asinchuk — APPROVED** [review]

**Pull request merged**